src/algebra/geom_sum.lean

```diff
@@ -172,6 +172,28 @@ theorem geom_sum₂_mul [comm_ring α] (x y : α) (n : ℕ) :
   (∑ i in range n, x ^ i * (y ^ (n - 1 - i))) * (x - y) = x ^ n - y ^ n :=
 (commute.all x y).geom_sum₂_mul n
 
+theorem sub_dvd_pow_sub_pow [comm_ring α] (x y : α) (n : ℕ) : x - y ∣ x^n - y^n :=
+  dvd.intro_left _ (geom_sum₂_mul x y n)
+
+theorem nat.sub_dvd_pow_sub_pow (x y n : ℕ) : x - y ∣ x^n - y^n :=
+begin
+  cases le_or_lt y x with h,
+  { have : y^n ≤ x^n := nat.pow_le_pow_of_le_left h _,
+    exact_mod_cast sub_dvd_pow_sub_pow (x : ℤ) ↑y n },
+  { have : x^n ≤ y^n := nat.pow_le_pow_of_le_left h.le _,
+    exact (nat.sub_eq_zero_of_le this).symm ▸ dvd_zero (x - y) }
+end
+
+theorem odd.add_dvd_pow_add_pow [comm_ring α] (x y : α) {n : ℕ} (h : odd n) : x + y ∣ x^n + y^n :=
+begin
+  have h₁ := geom_sum₂_mul x (-y) n,
+  rw [odd.neg_pow h y, sub_neg_eq_add, sub_neg_eq_add] at h₁,
+  exact dvd.intro_left _ h₁,
+end
+
+theorem nat.add_dvd_pow_add_pow_of_odd (x y : ℕ) {n : ℕ} (h : odd n) : x + y ∣ x^n + y^n :=
```

Comment:
Maybe a name that starts with `odd.` (but contains `nat`)? I think having dot notation is always a good thing.

Reply:
Perhaps `odd.nat_add_dvd_pow_add_pow`? Alternatively `odd.add_dvd_pow_add_pow'`, but I don't think the apostrophe is descriptive enough in that case.

Reply:
`odd.nat_add_dvd_pow_add_pow` looks good, thanks!

Reply:
In that case you should also change the name of the other lemma about `ℕ`.

Reply:
So rename `nat.sub_dvd_pow_sub_pow` to `nat_sub_dvd_pow_sub_pow`?

Reply:
Yes, just to be consistent. Maybe the other can be `odd.nat.add_dvd_pow_add_pow`, but please check that dot notation works.

Reply:
`#check (show odd 5, by dec_trivial).nat.add_dvd_pow_add_pow 3 4` gives me a syntax error, so I'd assume that it doesn't work. I'll just switch the first definition to `nat_sub_dvd_pow_sub_pow`.

Reply:
You need an additional `^` (as in `(show odd 5, by dec_trivial)^.nat.add_dvd_pow_add_pow 3 4`) to make it work.

By the way, no need to format commit messages carefully, they'll be squash merged into master anyway, with commit message from the PR title and description.